src/geometry/GeoTrackView.i.hh

```diff
@@ -67,9 +65,7 @@ CELER_FUNCTION GeoTrackView& GeoTrackView::operator=(const Initializer_t& init)
 }
 
 //---------------------------------------------------------------------------//
-/*!
- * Find the distance to the next geometric boundary.
- */
+//! Find the distance to the next geometric boundary.
 CELER_FUNCTION void GeoTrackView::find_next_step()
```

Comment:
As a side note, this routine (really ComputeStepAndPropagatedState) is usually only used in testing framework as during the simulation proper the step is limited by physics and we also want to get the value of the safety.

We will (unless I am missing something) also need a wrapper around FindNextBoundaryAndStepAndSafety.

Reply:
Is that true for neutral particles (straight line transport) as well? I thought that with the linear propagator, the "next step" mechanics could work. The linear propagator (when you give it the step limit imposed by the physics) should be responsible for stopping before the boundary if necessary, or moving to the next volume if not. Right?

Reply:
> The linear propagator (when you give it the step limit imposed by the physics) should be responsible for stopping before the boundary if necessary, or moving to the next volume if not. Right?

That's pretty much the purpose of FindNextBoundaryAndStepAndSafety (i.e. detect if there is boundary within the step limit and "return" the next volume if there is).